test/Microsoft.AspNet.Razor.Test/Parser/Html/HtmlAttributeTest.cs

```diff
@@ -14,6 +14,112 @@ namespace Microsoft.AspNet.Razor.Test.Parser.Html
 {
     public class HtmlAttributeTest : CsHtmlMarkupParserTestBase
     {
+        public static TheoryData SymbolBoundAttributeNames
+        {
+            get
+            {
+                return new TheoryData<string>
+                {
+                    "[item]",
+                    "[(item,",
+                    "(click)",
+                    "(^click)",
+                    "*something",
+                    "#local",
```

Comment:
Need to test more than just symbols Angular would allow. Non-Latin1 characters at least.

I also don't see tests verifying rejection of anything but whitespace and `=`.